checker.c

```diff
@@ -1,21 +1,52 @@
-#include <stdio.h>
-#include <assert.h>
+#include "checker.h"
+
+float readchargeRate(float chargeRate)
```

Comment:
method name does not indicate the functionality that it performs. To me it gives a meaning that parameter value is read and not returned.